usaspending_api/recipient/delta_models/recipient_profile.py

```diff
@@ -1,26 +1,330 @@
 RECIPIENT_PROFILE_COLUMNS = {
-    "recipient_level": "STRING NOT NULL",
-    "recipient_hash": "STRING",
-    "recipient_unique_id": "STRING",
-    "recipient_name": "STRING",
-    "recipient_affiliations": "ARRAY<STRING> NOT NULL",
-    "last_12_months": "numeric(23,2) NOT NULL",
-    "id": "LONG NOT NULL",
-    "last_12_contracts": "numeric(23,2) NOT NULL",
-    "last_12_direct_payments": "numeric(23,2) NOT NULL",
-    "last_12_grants": "numeric(23,2) NOT NULL",
-    "last_12_loans": "numeric(23,2) NOT NULL",
-    "last_12_months_count": "INTEGER NOT NULL",
-    "last_12_other": "numeric(23,2) NOT NULL",
-    "award_types": "ARRAY<STRING> NOT NULL",
-    "uei": "STRING",
-    "parent_uei": "STRING",
+    "recipient_level": {"delta": "STRING NOT NULL", "postgres": "TEXT NOT NULL"},
+    "recipient_hash": {"delta": "STRING", "postgres": "UUID"},
+    "recipient_unique_id": {"delta": "STRING", "postgres": "TEXT"},
+    "recipient_name": {"delta": "STRING", "postgres": "TEXT"},
+    "recipient_affiliations": {"delta": "ARRAY<STRING> NOT NULL", "postgres": "TEXT[]"},
+    "last_12_months": {"delta": "numeric(23,2) NOT NULL", "postgres": "numeric(23,2"},
+    "id": {"delta": "LONG NOT NULL", "postgres": "bigint"},
+    "last_12_contracts": {"delta": "numeric(23,2) NOT NULL", "postgres": "numeric(23,2"},
+    "last_12_direct_payments": {"delta": "numeric(23,2) NOT NULL", "postgres": "numeric(23,2"},
+    "last_12_grants": {"delta": "numeric(23,2) NOT NULL", "postgres": "numeric(23,2"},
+    "last_12_loans": {"delta": "numeric(23,2) NOT NULL", "postgres": "numeric(23,2"},
+    "last_12_months_count": {"delta": "INTEGER NOT NULL", "postgres": "numeric(23,2"},
+    "last_12_other": {"delta": "numeric(23,2) NOT NULL", "postgres": "numeric(23,2"},
+    "award_types": {"delta": "ARRAY<STRING> NOT NULL", "postgres": "TEXT[]"},
+    "uei": {"delta": "STRING", "postgres": "text"},
+    "parent_uei": {"delta": "STRING", "postgres": "text"},
 }
-
-recipient_profile_sql_string = rf"""
+RECIPIENT_PROFILE_DELTA_COLUMNS = {k: v["delta"] for k, v in RECIPIENT_PROFILE_COLUMNS.items()}
+RECIPIENT_PROFILE_POSTGRES_COLUMNS = {k: v["postgres"] for k, v in RECIPIENT_PROFILE_COLUMNS.items()}
+recipient_profile_create_sql_string = f"""
     CREATE OR REPLACE TABLE {{DESTINATION_TABLE}} (
-        {", ".join([f'{key} {val}' for key, val in RECIPIENT_PROFILE_COLUMNS.items()])}
+        {", ".join([f'{key} {val}' for key, val in RECIPIENT_PROFILE_DELTA_COLUMNS.items()])}
     )
     USING DELTA
     LOCATION 's3a://{{SPARK_S3_BUCKET}}/{{DELTA_LAKE_S3_PATH}}/{{DESTINATION_DATABASE}}/{{DESTINATION_TABLE}}'
     """
+recipient_profile_load_sql_strings = [
+    f"""
+        DELETE FROM {{DESTINATION_DATABASE}}.{{DESTINATION_TABLE}};
```

Comment:
Seeing the whole review, there's not one major INSERT, but a lot of MERGEs. I think you could still save this delete step and the first INSERT be an `INSERT OVERWRITE` which implicitly performs a TRUNCATE. But not sure if it will make much difference in speed and may make the code less clear.

Reply:
refactored